lib/mix/lib/mix/tasks/xref.ex

```diff
@@ -956,17 +971,21 @@ defmodule Mix.Tasks.Xref do
           {roots, callback, count} =
             roots_and_callback(file_references, filter, sources, sinks, opts)
 
-          path = Keyword.get(opts, :output, "xref_graph.dot")
+          file_spec = Mix.Utils.write_dot_graph!("xref graph", Enum.sort(roots), callback, opts)
 
-          Mix.Utils.write_dot_graph!(path, "xref graph", Enum.sort(roots), callback, opts)
+          if file_spec != "-" do
+            png_file_spec = (file_spec |> Path.rootname() |> Path.basename()) <> ".png"
 
-          if path != "-" do
-            png_path = (path |> Path.rootname() |> Path.basename()) <> ".png"
+            location =
+              if Path.dirname(file_spec) == "." and !String.starts_with?(file_spec, "."),
+                do: " in the current directory",
+                else: ""
 
+            # maybe this would be better as a comment in the dot file.
             """
-            Generated #{inspect(path)} in the current directory. To generate a PNG:
+            Generated '#{inspect(file_spec)}'#{location}. To generate a PNG:
```

Comment:
My suggestion would be to simply do:

```suggestion
            Generated #{inspect(Path.relative_to_cwd(file_spec)}. To generate a PNG:
```

So we no longer need to worry about saying current directory or not. If it is in the current directory, we will shot a relative path, otherwise, absolute.

Reply:
ack. ill update as requested.